hudi-aws/src/main/java/org/apache/hudi/config/GlueCatalogSyncClientConfig.java

```diff
@@ -40,6 +42,28 @@ public class GlueCatalogSyncClientConfig extends HoodieConfig {
       .sinceVersion("0.14.0")
       .withDocumentation("Glue catalog sync based client will skip archiving the table version if this config is set to true");
 
+  public static final ConfigProperty<Integer> ALL_PARTITIONS_READ_PARALLELISM = ConfigProperty
+      .key(GLUE_CLIENT_PROPERTY_PREFIX + "all_partitions_read_parallelism")
+      .defaultValue(1)
+      .markAdvanced()
+      .withValidValues(IntStream.rangeClosed(1, 10).mapToObj(Integer::toString).toArray(String[]::new))
+      .sinceVersion("1.0.0")
+      .withDocumentation("Parallelism for listing all partitions(first time sync). Should be in interval [1, 10].");
```

Comment:
```suggestion
      .withDocumentation("Parallelism for listing all partitions(first time sync). Should be in the range of [1, 10].");
```

Reply:
It's a hard limit of 10 segments per AWS doc 
https://docs.aws.amazon.com/glue/latest/webapi/API_Segment.html